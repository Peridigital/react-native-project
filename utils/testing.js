import ShallowRenderer from 'react-test-renderer/shallow'

/**
 * Creates a shallow rendered string representation of a given react component
 *
 * @param {any} component - A React component to be snapshotted ex: <Text>Hello World<Text/>
 *
 * @returns {string} A shallow rendered string representation of the React component
 */
export function shallow (component) {
  const renderer = new ShallowRenderer()
  renderer.render(component)
  return renderer.getRenderOutput()
}
