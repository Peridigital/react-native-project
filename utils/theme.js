const colors = {
  primaryBackground: '#0C0D12',
  secondaryBackground: '#151821',
  primaryText: '#EAEAEA',
  border: '#7E4033',
  borderHighlight: '#D0A499'
}

export default colors
