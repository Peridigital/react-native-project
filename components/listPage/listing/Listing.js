import React from 'react'
import { noop } from 'lodash'
import {
  View,
  Text,
  Image,
  Pressable
} from 'react-native'
import PropTypes from 'prop-types'
import style from './Listing.style'
import { LinearGradient } from 'expo-linear-gradient'
import theme from '../../../utils/theme'

export function Listing ({ starship, navigateToDetails, hasFontsLoaded }) {
  return (
    <Pressable
      onPress={navigateToDetails}
    >
      <View style={style.listingBorder}>
      <View style={style.listingFlair} />
        <View style={style.listing}>
            <Image style={style.shipImage} source={{ uri: 'https://static.wikia.nocookie.net/starwars/images/2/24/TantiveIV-TSWB.png/revision/latest?cb=20221006055135' }}/>
            <LinearGradient start={{ x: 0, y: 0 }} end={{ x: 1, y: 0 }} style={style.gradientBackground} colors={[theme.primaryBackground, 'rgba(0,0,0,0)']} />
            {hasFontsLoaded ? <Text style={{ fontFamily: 'Orbitron_400Regular', ...style.title }}>{starship?.name}</Text> : <></>}
        </View>

      </View>
    </Pressable>
  )
}

Listing.propTypes = {
  starship: PropTypes.object,
  navigateToDetails: PropTypes.func,
  hasFontsLoaded: PropTypes.bool
}

Listing.defaultProps = {
  navigateToDetails: noop,
  hasFontsLoaded: false
}

export default Listing
