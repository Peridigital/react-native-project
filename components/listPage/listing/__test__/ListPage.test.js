/* eslint-disable no-undef */
import React from 'react'
import { shallow } from '../../../../utils/testing'
import Listing from '../Listing'

describe('Listing', () => {
  it('should render', () => {
    const wrapper = shallow(<Listing />)

    expect(wrapper).toMatchSnapshot()
  })
})
