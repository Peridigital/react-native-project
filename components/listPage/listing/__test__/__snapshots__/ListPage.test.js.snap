// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing should render 1`] = `
<View>
  <Memo(Pressable)
    onPress={[Function]}
  >
    <Text />
  </Memo(Pressable)>
</View>
`;
